### PR TITLE
UCT/DC: HW TM support

### PR DESCRIPTION
**Pull request by alex-mikheev**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4632/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4647/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4652/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/hpc-ucx-pr__LAB_sourced/4665/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4674/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4675/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4680/ for details (Mellanox internal link).

File: src/uct/ib/dc/verbs/dc_verbs.c [hunk 1]
```diff
@@ -749,6 +774,316 @@ static unsigned uct_dc_verbs_iface_progress(uct_iface_h tl_iface)
...
+        }
+    }
+
+    arm_rx_solicited = 0;
+    arm_rx_all       = 0;
+    if (events & UCT_EVENT_RECV_AM) {
+        arm_rx_solicited = 1; /* to wake up on active messages */
+    }
+    if (((events & UCT_EVENT_SEND_COMP) && rc_iface->config.fc_enabled) ||
+         UCT_RC_VERBS_TM_ENABLED(&iface->verbs_common)) {
```
Comment (yosefe) [inline src/uct/ib/dc/verbs/dc_verbs.c]: need to extract common function e.g uct_rc_iface_common_event_arm(rc_iface, events, force_rx_all)
rc will use force_rx_all=0 and dc will use force_rx_all=1 if TM is enabled

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4714/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4713/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
```
15:18:49 /scrap/jenkins/workspace/hpc-ucx-pr/label/r-vmb-fedora18-x86-64/worker/0/contrib/../src/uct/ib/dc/verbs/dc_verbs.c: In function ‘uct_dc_verbs_iface_event_arm’:
15:18:49 /scrap/jenkins/workspace/hpc-ucx-pr/label/r-vmb-fedora18-x86-64/worker/0/contrib/../src/uct/ib/dc/verbs/dc_verbs.c:1130:27: error: unused variable ‘iface’ [-Werror=unused-variable]
15:18:49 cc1: all warnings being treated as errors
15:18:49 make[2]: *** [ib/dc/verbs/libuct_la-dc_verbs.lo] Error 1
```

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4715/ for details (Mellanox internal link).

**Pull request merged**